src/poetry_plugin_export/exporter.py

```diff
@@ -71,7 +75,15 @@ def export(self, fmt: str, cwd: Path, output: IO | str) -> None:
 
         getattr(self, self.EXPORT_METHODS[fmt])(cwd, output)
 
+    def _export_constraints_txt(self, cwd: Path, output: IO | str) -> None:
+        return self._export_generic_txt(cwd, output, export_as_constraints=True)
+
     def _export_requirements_txt(self, cwd: Path, output: IO | str) -> None:
+        return self._export_generic_txt(cwd, output, export_as_constraints=False)
+
+    def _export_generic_txt(
```

Comment:
I would rather see explicit `with_extras` and `allow_editable` here vs. `export_as_constraints`.

Reply:
Also, looking at this further, I think it would be more clear to implement these parameters as part of a [partially applied function](https://docs.python.org/3/library/functools.html#functools.partial) -- which could be stored directly in the table and remove the need for `getattr`.

Reply:
> Also needs a test to make sure that we raise an exception when an editable dependency is exported in constraints.txt mode

The `if package.develop:` branch has no test coverage in the main branch. I tried setting `develop = True` in various places in the mock lock data in the tests, but none seemed to trigger that code. If you think adding tests to the exception raising code I added is important, any guidance on how to do this would be much appreciated!

> I would rather see explicit `with_extras` and `allow_editable` here vs. `export_as_constraints`.

:+1:

> Also, looking at this further, I think it would be more clear to implement these parameters as part of a [partially applied method](https://docs.python.org/3/library/functools.html#functools.partialmethod) -- which could be stored directly in the table and remove the need for `getattr`.

I may have misunderstood you, but I didn't see a way to use the return values of `partialmethod` directly in the table. Instead, I used `functools.partial` and then manually add the `self` argument in the `export` function.

Reply:
I forgot `partialmethod` returns a descriptor. I've pushed a commit that implements this symmetrically with the prior code, using partialmethod.